src/app/courses-page/services/courses.service.ts

```diff
@@ -41,8 +41,15 @@ export class CoursesService {
     return this.courses;
   }
 
-  createCourse(course: Course): void {
-    this.courses.push(course);
+  createCourse(course: Partial<Course>): void {
+    const updatedCourse = {
+      id: this.courses[this.courses.length - 1].id + 1,
```

Comment:
do not rely on array length, you will end up with two same ids if you add a course, delete some other and add new one
here is the package for id generation, you can use this)
https://www.npmjs.com/package/uuid

In idead world id should be assigned by backend

Reply:
@SteveRusin uuid is used. Thanks)